spec/models/review_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe Review, type: :model do
+  describe 'validations' do
+    let(:user) { create(:user) }
+    let(:product) { create(:product) }
+
+    it 'is valid with valid attributes' do
+      review = build(:review, user:, product:, rating: 4)
+      expect(review).to be_valid
+    end
+
+    it 'is not valid without a user' do
+      review = build(:review, user: nil)
+      expect(review).to_not be_valid
+      expect(review.errors[:user]).to include('must exist')
+    end
+
+    it 'is not valid without a product' do
+      review = build(:review, product: nil)
+      expect(review).to_not be_valid
+      expect(review.errors[:product]).to include('must exist')
+    end
+
+    it 'is not valid without a rating' do
+      review = build(:review, rating: nil)
+      expect(review).to_not be_valid
+      expect(review.errors[:rating]).to include("can't be blank")
+    end
+
+    it 'is not valid with a rating less than 1' do
+      review = build(:review, rating: 0)
+      expect(review).to_not be_valid
+      expect(review.errors[:rating]).to include('is not included in the list')
+    end
+
+    it 'is not valid with a rating greater than 5' do
+      review = build(:review, rating: 6)
+      expect(review).to_not be_valid
+      expect(review.errors[:rating]).to include('is not included in the list')
+    end
```

Comment:
add a context for rating related specs inside "describe validation"

Reply:
Done bhai